components/sup/src/manager/service/mod.rs

```diff
@@ -928,10 +945,59 @@ impl Service {
                 (false, _) => HealthCheck::Critical,
             }
         };
-        self.last_health_check = Some(Instant::now());
+
+        // We have just ran a check; therefore we must unset the next scheduled check time
+        // in anticipation of `None` value being used in the next scheduled check time calculation.
+        self.scheduled_health_check = None;
+
+        if check_result == HealthCheck::Ok {
+            self.schedule_routine_health_check();
+            debug!(
+                "Service ({}) health check is: {}",
+                self.spec_ident, check_result
+            );
+        } else {
+            self.schedule_special_health_check();
+            debug!(
+                "Service ({}) health check is: {}. Re-running within default {:#?}",
+                self.spec_ident,
+                check_result,
+                HealthCheckInterval::default()
+            );
```

Comment:
As is, the logging would look like
```
DEBUG 2018-12-19T15:46:51Z: habitat_sup::manager::service: Scheduling next health check for (foo) in (30s)
DEBUG 2018-12-19T15:46:51Z: habitat_sup::manager::service: Service (foo) health check is: CRITICAL. Re-running within default HealthCheckInterval(
    30s
)
```
I'd suggest moving the `debug!` here above the call to `self.schedule_special_health_check` and changing it to:
```rust
            debug!(
                "Service ({}) health check is: {}; scheduling special health check",
                self.spec_ident, check_result
            );
```
giving nicer logging:
```
DEBUG 2018-12-19T15:46:51Z: habitat_sup::manager::service: Service (foo) health check is: CRITICAL; scheduling special health check
DEBUG 2018-12-19T15:46:51Z: habitat_sup::manager::service: Scheduling next health check for (foo) in (30s)
```

Reply:
That works for me! See this last [commit](https://github.com/habitat-sh/habitat/pull/5890/commits/0b33755fd040ec33da4b2e7c85e8faa0d4bd6436)